code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -131,11 +132,10 @@
 
 /mob/living/carbon/Xenomorph/proc/handle_critical_health_updates()
 	var/turf/T = loc
-	if(istype(T))
-		if(!locate(/obj/effect/alien/weeds) in T) //In crit, damage is maximal if you're caught off weeds
-			adjustBruteLoss(2.5 - warding_aura*0.5) //Warding can heavily lower the impact of bleedout. Halved at 2.5 phero, stopped at 5 phero
-		else
-			adjustBruteLoss(-warding_aura*0.5) //Warding pheromones provides 0.25 HP per second per step, up to 2.5 HP per tick.
+	if((istype(T) && locate(/obj/effect/alien/weeds) in T))
+		heal_wounds(warding_aura*0.5) //Warding pheromones provides 0.25 HP per second per step, up to 2.5 HP per tick.
+	else if(!(xeno_caste.caste_flags & CASTE_INNATE_HEALING))
```

Comment:
We do want the innate healing castes, such as larvas, to bleed out. It `if` part can be left out.